pkg/subctl/lighthouse/deploy/ensure.go

```diff
@@ -17,22 +17,77 @@ limitations under the License.
 package lighthouse
 
 import (
+	"encoding/json"
+	"errors"
 	"fmt"
+	"os/exec"
+	"strings"
 
+	"github.com/spf13/cobra"
 	"k8s.io/client-go/rest"
 
+	"github.com/kubernetes-sigs/kubefed/pkg/version"
 	"github.com/submariner-io/submariner-operator/pkg/internal/cli"
 	lighthousedns "github.com/submariner-io/submariner-operator/pkg/subctl/lighthouse/dns"
 	"github.com/submariner-io/submariner-operator/pkg/subctl/lighthouse/install"
 	"github.com/submariner-io/submariner-operator/pkg/subctl/operator/kubefed"
 )
 
 const (
-	DefaultControllerImageName    = "lighthouse-controller"
-	DefaultControllerImageRepo    = "quay.io/submariner"
-	DefaultControllerImageVersion = "0.0.1"
+	defaultControllerImageName    = "lighthouse-controller"
+	defaultControllerImageRepo    = "quay.io/submariner"
+	defaultControllerImageVersion = "0.0.1"
+	kubefedVersion                = "v0.1.0-rc3"
+	kubefedctlVersionPrefix       = "kubefedctl version: version.Info"
 )
 
+var (
+	serviceDiscovery bool
+	imageRepo        string
+	imageVersion     string
+)
+
+func AddFlags(cmd *cobra.Command, prefix string) {
+	cmd.PersistentFlags().BoolVar(&serviceDiscovery, prefix, false,
+		"Enable Multi Cluster Service Discovery")
+	cmd.PersistentFlags().StringVar(&imageRepo, prefix+"-repo", defaultControllerImageRepo,
+		"Service Discovery Image repository")
+	cmd.PersistentFlags().StringVar(&imageVersion, prefix+"-version", defaultControllerImageVersion,
+		"Service Discovery Image version")
+}
+
+func Validate() error {
+	if serviceDiscovery {
+		// Check we can run kubefedctl, and it’s the right version
+		out, err := exec.Command("kubefedctl", "version").CombinedOutput()
+		if err != nil {
+			return err
+		}
+		output := string(out)
+		if !strings.HasPrefix(output, kubefedctlVersionPrefix) {
+			return fmt.Errorf("unable to determine kubefedctl version, please use %s", kubefedVersion)
+		}
+		// We need to tweak the output to make it valid JSON again...
+		output = output[len(kubefedctlVersionPrefix):]
+		output = strings.Replace(output, "Version", "\"gitVersion\"", 1)
+		output = strings.Replace(output, "GitCommit", "\"gitCommit\"", 1)
+		output = strings.Replace(output, "GitTreeState", "\"gitTreeState\"", 1)
+		output = strings.Replace(output, "BuildDate", "\"buildDate\"", 1)
+		output = strings.Replace(output, "GoVersion", "\"goVersion\"", 1)
+		output = strings.Replace(output, "Compiler", "\"compiler\"", 1)
+		output = strings.Replace(output, "Platform", "\"platform\"", 1)
+		v := version.Info{}
+		err = json.Unmarshal([]byte(output), &v)
+		if err != nil {
+			return err
+		}
+		if v.Version != kubefedVersion {
+			return fmt.Errorf("invalid kubefedctl version %s, please use %s", v.Version, kubefedVersion)
+		}
+	}
+	return errors.New("")
```

Comment:
why not return nil?

Reply:
Hah, thanks, I was using that to check the new code and forgot to remove it!